src/bin/tikv-ctl.rs

```diff
@@ -63,19 +67,36 @@ fn new_debug_executor(
     db: Option<&str>,
     raft_db: Option<&str>,
     host: Option<&str>,
+    cfg_path: Option<&str>,
     mgr: Arc<SecurityManager>,
 ) -> Box<DebugExecutor> {
     match (host, db) {
         (None, Some(kv_path)) => {
-            let db = util::rocksdb::open(kv_path, ALL_CFS).unwrap();
-            let raft_db = if let Some(raft_path) = raft_db {
-                util::rocksdb::open(raft_path, &[CF_DEFAULT]).unwrap()
-            } else {
-                let raft_path = PathBuf::from(kv_path).join("../raft");
-                util::rocksdb::open(raft_path.to_str().unwrap(), &[CF_DEFAULT]).unwrap()
-            };
-            Box::new(Debugger::new(Engines::new(Arc::new(db), Arc::new(raft_db)))) as
-                Box<DebugExecutor>
+            let cfg = cfg_path.map_or_else(TiKvConfig::default, |path| {
+                File::open(&path)
+                    .and_then(|mut f| {
+                        let mut s = String::new();
+                        f.read_to_string(&mut s).unwrap();
+                        let c = toml::from_str(&s).unwrap();
+                        Ok(c)
+                    })
+                    .unwrap()
+            });
+            let kv_db_opts = cfg.rocksdb.build_opt();
+            let kv_cfs_opts = cfg.rocksdb.build_cf_opts();
+            let kv_db = rocksdb_util::new_engine_opt(kv_path, kv_db_opts, kv_cfs_opts).unwrap();
```

Comment:
The old `open` function is used by other place or not? If it's only for test, we can put it into `cfg[(test)]` block.